client/src/data/interviews/templateValidator.ts

```diff
@@ -0,0 +1,58 @@
+import { InterviewTemplate, Question, QuestionCategory } from '@client/data/interviews/types';
+
+export class TemplateValidationError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = 'TemplateValidationError';
+  }
+}
+
+export class InterviewTemplateValidator {
```

Comment:
IMO we don't really need classes here, it can be just a function that performs checks and throws error if something is wrong